packages/block-library/src/navigation-link/index.php

```diff
@@ -20,7 +20,7 @@ function block_core_navigation_link_build_css_colors( $context ) {
 
 	// Text color.
 	$has_named_text_color  = array_key_exists( 'textColor', $context );
-	$has_custom_text_color = array_key_exists( 'customTextColor', $context );
+	$has_custom_text_color = isset( $context['style']['color']['text'] );
```

Comment:
General comment that doesn't require any action in this PR—it'd be nice if there was a php function for handling the colors support setting.

Similarly, the amount of wrangling that has to be done of colors passed through context feels like something that could be improved.